### PR TITLE
Convert from yarn to npm #9282

### PR DESCRIPTION
**Pull request by chrabyrd**
### Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Issues Solved

#9282

### CODE REVIEW

File: docker/Readme.md [hunk 1]
```diff
@@ -315,10 +315,10 @@ The volume section at **point 4** also mounts a `settings_local.py` into the con
...
 
-##### Yarn components
+##### npm components
 Because your volume from `step 4` is laid over the files in your container, files generated during image `build` are hidden.  
 In order to install newly added static file packages, run the following command:  
-    `docker-compose -f .\docker-compose-local.yml run arches install_yarn_components`
+    `docker-compose -f .\docker-compose-local.yml run arches install_npm_components`
 
 ##### docker-compose-local.yml
 Here is a link to a working [docker-compose-local.yml](https://gist.github.com/jmunowitch/c63fa39be4651b9bf2f0b1abc69f7479) file
```
Comment (jacobtylerwalls) [inline docker/Readme.md:324]: Let's delete this, it doesn't even resolve to anything

File: package.json [hunk 2]
```diff
@@ -102,15 +97,14 @@
...
-        "select2": "node_modules/select2/select2",
         "select-woo": "node_modules/select-woo/dist/js/selectWoo.full",
         "select-woo-src": "node_modules/select-woo/src/js/select2",
         "bootstrap": "node_modules/bootstrap/dist/js/bootstrap.min",
         "jquery-ui": "node_modules/jqueryui/jquery-ui.min",
         "backbone": "node_modules/backbone/backbone-min",
         "underscore": "node_modules/underscore/underscore-min",
         "jquery-validate": "node_modules/jquery-validation/dist/jquery.validate.min",
-        "d3": "node_modules/d3/dist/d3.min",
+        "d3": "node_modules/d3/src/index.js",
```
Comment (jacobtylerwalls) [inline package.json:107]: I'm unable to build the frontend unless I revert this change. Do you know why?

**Review by jacobtylerwalls — CHANGES_REQUESTED** [review]
d3 caused a problem with my build, do you see the same?

File: docker/Readme.md (hunk 1 above)
Comment (chrabyrd) [reply docker/Readme.md:324]: could you clarify? `install_npm_components` matches to a function declared in `entrypoint.sh`. Admittedly I don't run docker so my context is weak

File: package.json (hunk 2 above)
Comment (chrabyrd) [reply package.json:107]: hmm just checked this and it's working for me -- and breaks the other way. Did you update the project's `package.json` to point to this branch?

```
    "dependencies": {
        "arches": "archesproject/arches#cbyrd-9282-convert-from-yarn-to-npm"
    },
```

File: docker/Readme.md (hunk 1 above)
Comment (jacobtylerwalls) [reply docker/Readme.md:324]: no worries -- I meant the comment right below your change, github couldn't let me comment directly on an unchanged line. the gist.github.com/... link goes nowhere

File: package.json (hunk 2 above)
Comment (jacobtylerwalls) [reply package.json:107]: whoops, I definitely missed that--lgtm!

**Review by jacobtylerwalls — APPROVED** [review]
LGTM, sending over to Aaron to have a look.

**Review by aarongundel — APPROVED** [review]
This is really good to see.  A couple of notes from my testing (that you already saw) - relative pathing in the package.json file will break with npm because of the hoisting differences between yarn and npm.  Our workaround is to avoid using relative pathing in the package.json.  I am not sure if this is worth documenting or not, since it's not really a solution.  

I did notice performance impacts when doing a full build of arches vgm.  Not too surprising, but noteworthy.  

You may want to note that if someone is using a container based setup, they'll need to update any node references therein.  For instance, my entrypoint script for vgm was calling yarn install directly.  Some other projects will likely be impacted by that change.  It could end up being the case if that if you're not diligent about removing yarn from your container setup, you'd have dueling package managers - and that'll probably be a receipe for a forum post.  

Other than that, looks good!  Your call on documentation updates.
